cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -231,6 +231,9 @@ public void update(HttpRequest request, HttpResponder responder, @PathParam("nam
   private DatasetInstanceConfiguration  getInstanceConfiguration(HttpRequest request) {
     Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()), Charsets.UTF_8);
     DatasetInstanceConfiguration creationProperties = GSON.fromJson(reader, DatasetInstanceConfiguration.class);
+    if (creationProperties.getProperties() == null) {
```

Comment:
I don't like having extra logic modifying the deserialized object after a `GSON.fromJson` since if we want to deserialize `DatasetInstanceConfiguration` elsewhere, we may have to copy paste this logic there. Even if not, it would be confusing since there's different logic in deserializing `DatasetInstanceConfiguration` in different places, which would make things messy.

Instead, if you want to have GSON use a default value in case a field is missing, you can probably initialize default fields in the default constructor in `DatasetInstanceConfiguration`. Alternatively, you can use something like `FlowletSpecificationCodec` then configure your GSON instance to use it.